src/main/java/com/lucidworks/spark/SolrRelation.java

```diff
@@ -51,14 +61,23 @@ public SolrRelation(SQLContext sqlContext, Map<String,String> config, DataFrame
       throw new IllegalArgumentException("SQLContext cannot be null!");
 
     this.sqlContext = sqlContext;
-
+    this.sc =  new JavaSparkContext(sqlContext.sparkContext());
+    if (optionalParam(config, PRESERVE_SCHEMA, "N").equals("Y")) {
+      preserveSchema = true;
+    };
     String zkHost = requiredParam(config, SOLR_ZK_HOST_PARAM);
     String collection = requiredParam(config, SOLR_COLLECTION_PARAM);
     String query = optionalParam(config, SOLR_QUERY_PARAM, "*:*");
     splitFieldName = optionalParam(config, SOLR_SPLIT_FIELD_PARAM, null);
     if (splitFieldName != null)
       splitsPerShard = Integer.parseInt(optionalParam(config, SOLR_SPLITS_PER_SHARD_PARAM, "10"));
-    solrRDD = new SolrRDD(zkHost, collection);
+    if(!preserveSchema) {
```

Comment:
Please use consistent style by putting a space after the if keyword before the opening parens, i.e. 

if (!preserveSchema)